gdnative-core/src/nativescript/class.rs

```diff
@@ -96,6 +96,7 @@ pub trait NativeClass: Sized + 'static {
     where
         Self::Base: Instanciable,
     {
+        debug_assert!(is_class_registered::<Self>(), "`NativeClass` {type_name} has not been added in init function; please ensure that you call `handle.add_class::<{type_name}>()` in function callback in `godot_nativescript_init` macro", type_name = std::any::type_name::<Self>());
```

Comment:
Why isn't this in `Instance::maybe_emplace` instead? The assertion should catch more cases there. The message is a bit convoluted here, too. Perhaps consider: `` `{type_name}` must be registered before it can be used; call `handle.add_class::<{type_name}>()` in your `nativescript_init` callback ``?

Reply:
Sorry about that! I didn't realize that `Instance::maybe_emplace` was a more broadly called function. I'll move the check into that method instead. Thank you for the great feedback!